### PR TITLE
updated ReadMe, adding test and files for task 3

### PR DESCRIPTION
**Pull request by rsdecoder**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,4 +6,7 @@
 3. Set PGDATABASE = nc_news_test inside .env.test file
 
 ----------------
-4. Completed task 2 - GET api-topics
\ No newline at end of file
+4. Completed task 2 - GET api-topics
+5. Completed task 3 - GET -api. I have written code for two ways of doing this.
+    1 -> Requiring endpoints.json and just returning that object.
+    2 -> By using fs.readFile and returning the parsed objects. This was my first way of doing it and have commented it out. Both of these passed my tests. Just wondering if both are ok or if the first one is still the best one. I like the first one better as the code is less. Please give me feedback on this. I will delete the commented/unnecessary code once after the feedback is given. Thank you.
```
Comment (ghost) [inline README.md:12]: This is not the best way to communicate with the tutors regarding your approach, keep your readme for instructions for those using your project

File: __tests__/app.test.js [hunk 2]
```diff
@@ -38,4 +39,17 @@ describe("GET", () => {
...
+  describe('/api', () => {
+    test('STATUS - 200, should respond with all the list of endpoints available ', () => {
+        return request(app)
+        .get("/api")
+        .expect(200)
+        .then(({body}) => {
+          console.log(body)
+          const bodyLength = Object.keys(body).length
+          expect(bodyLength).toBe(3)
+          expect(body).toEqual(endpoints)
```
Comment (ghost) [inline __tests__/app.test.js:51]: body toEqual the endpoints json is the better way here, there is no need for line 49 or 50. If you're worried its too short, don't be.

File: controllers/errors.controllers.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const express = require("express");
+const app = express();
+
+app.use((err, request, response, next) => {
+    console.log('err in error handling middleware', err)
+    if(err) {
+        response.status(500).send({msg: 'Internal server error'})
+    }
+    next(err)
+})
```
Comment (ghost) [inline controllers/errors.controllers.js:10]: What is going on in this file? This is not how to set up an independent error handling file, refer back to the notes on this. No need to make a new express in here.

File: models/app.models.js [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
-module.exports = selectAllTopics;
\ No newline at end of file
+// function selectAllApis () {
+//     return fs.readFile('./endpoints.json')
+//     .then((fileContents) => {
+//         const parsedFileContents = JSON.parse(fileContents)
+//         return parsedFileContents;
+//     })
+// }
+
```
Comment (ghost) [inline models/app.models.js:11]: This file is doing nothing and shouldn't really be here.

File: controllers/app.controllers.js [hunk 5]
```diff
@@ -1,21 +1,21 @@
...
+// function getAllApis (request, response) {
+    
+//     selectAllApis().then((endpoints)=> { 
+//         response.status(200).send(endpoints)
+        
+//     })
+// }
+function getAllApis (request, response) {
+    
+    response.status(200).send(endpoints)
```
Comment (ghost) [inline controllers/app.controllers.js:18]: This is the better way of handling this 👍

**Review by ghost — COMMENTED** [review]
Some comments for you to read here. It is much better to just require in the endpoints.json and serve it right to the client without needing to read it :)

File: controllers/errors.controllers.js (hunk 3 above)
Comment (rsdecoder) [reply controllers/errors.controllers.js:10]: I wanted to move error handling into another file after I learnt it from lecture this morning.  I had corrected it this afternoon. At least I think so.

File: models/app.models.js (hunk 4 above)
Comment (rsdecoder) [reply models/app.models.js:11]: Was separating concerns and had to move things into their independent files. will remove it.

**Pull request merged**